packages/frontend/src/hooks/useIsCreatorAllowed.ts

```diff
@@ -0,0 +1,30 @@
+import type { Address } from "viem";
+import { useChainId, useContractReads } from "wagmi";
+import { CHAIN_ADDRESSES, FACTORY_ABI, ChainId } from "@carrot-kpi/sdk";
+
+export const useIsCreatorAllowed = (creator?: Address) => {
+    const chainId = useChainId();
+
+    const factoryAddress = CHAIN_ADDRESSES[chainId as ChainId].factory;
+    const { data, isLoading } = useContractReads({
+        contracts: [
+            {
+                address: factoryAddress,
+                abi: FACTORY_ABI,
+                functionName: "permissionless",
+            },
+            {
+                address: factoryAddress,
+                abi: FACTORY_ABI,
+                functionName: "creatorAllowed",
+                args: [creator as Address],
+            },
+        ],
+        enabled: chainId in ChainId,
```

Comment:
Should we add `creator` in the condition here?

Reply:
Addressed in a957f5af4b51ebbb78297cca3d341884afcefd2e